xmodule/video_block/video_block.py

```diff
@@ -357,11 +357,8 @@ def get_html(self, view=STUDENT_VIEW, context=None):  # lint-amnesty, pylint: di
         # for it, we fall back on whatever we find in the VideoBlock.
         if not download_video_link and self.download_video:
             if self.html5_sources:
-                download_video_link = self.html5_sources[0]
-
-            # don't give the option to download HLS video urls
-            if download_video_link and download_video_link.endswith('.m3u8'):
-                download_video_link = None
+                # If there are multiple html5 sources, we use the first non HLS video urls
+                download_video_link = next(filter(lambda vid: not vid.endswith('.m3u8'), self.html5_sources), None)
```

Comment:
```suggestion
                download_video_link = next((url for url in self.html5_sources if not vid.endswith('.m3u8')),  None)
```

Reply:
@sathiscode I think filter+lambda is probably a overkill for a simple if check. Do you think we can simplify it as I have suggested above?

Reply:
Yes, certainly its much simplter. Will simplify it.